### PR TITLE
TDD: check attemps in real word

### PR DESCRIPTION
**Pull request by badrbmb**

### CODE REVIEW

File: longest_word/game.py [hunk 1]
```diff
@@ -6,8 +6,9 @@
 
 import random
 import string
+import requests
 
-
+URL = "https://wagon-dictionary.herokuapp.com/"
```
Comment (gmanchon) [inline longest_word/game.py]: why not put this as a class constant ?

File: longest_word/game.py [hunk 2]
```diff
@@ -19,4 +20,12 @@ def is_valid(self, word: str) -> bool:
         """Check if a given world is valid"""
         if not isinstance(word, str):
             return False
-        return all(x in self.grid for x in list(word))
+        # check if the given  world is a subset of the desired solution
+        if not all(x in self.grid for x in list(word)):
```
Comment (gmanchon) [inline longest_word/game.py]: do we need to cast into a list ? ;)

Comment (gmanchon) [reply longest_word/game.py]: does this handle individual letter count ? ( "GROOVE" invalid for grid "GROVEABC" )

File: tests/test_game.py [hunk 3]
```diff
@@ -16,13 +16,13 @@ def test_game_initialization(self):
...
             # setup
             game = Game()
             # exercise
             game.grid = test_grid
             # verifiy
-            test_word = 'DATAENGINEERING'
+            test_word = 'OVERDOSE'
             assert game.is_valid(test_word) is True
 
     def test_is_not_valid(self):
```
Comment (gmanchon) [inline tests/test_game.py:40]:
```suggestion
   def test_letter_count(self):

            test_grid = list('GROVEABC')
            # setup
            game = Game()
            # exercise
            game.grid = test_grid
            # verifiy
            test_word = 'GROOVE'
            assert game.is_valid(test_word) is False

    def test_is_not_valid(self):
```

File: longest_word/game.py (hunk 2 above)
Comment (badrbmb) [reply longest_word/game.py]: changed the implementation to match proposed solution for this :p

File: tests/test_game.py (hunk 3 above)
Comment (badrbmb) [reply tests/test_game.py:40]: thx addedd additional test for case like this

**Review by gmanchon — APPROVED** [review]
LGTM 👌

**Comment by badrbmb** [comment]
Added real word check using lewagon internal word dictionary API

**Pull request merged**